packages/web-frontend/src/tests/utils/periodGranularities.test.js

```diff
@@ -0,0 +1,169 @@
+        expect(endDate.format()).toEqual('2019-02-04T23:59:59+11:00');
+      });
+
+      it('works when only end date provided', () => {
+        const { startDate, endDate } = getDefaultDates({
+          periodGranularity: 'one_day_at_a_time',
+          defaultTimePeriod: {
+            end: { unit: 'day', offset: -2 },
+          },
+        });
+        expect(startDate.format()).toEqual('2019-02-03T00:00:00+11:00');
+        expect(endDate.format()).toEqual('2019-02-03T23:59:59+11:00');
+      });
+
+      it('shorthand syntax', () => {
+        /*
+         * Equivalent to:
+         *   defaultTimePeriod: {
+         *     start: { unit: 'day', offset: -1 },
+         *   }
+         */
+        const { startDate, endDate } = getDefaultDates({
+          periodGranularity: 'one_day_at_a_time',
+          defaultTimePeriod: {
+            unit: 'day',
+            offset: -1,
+          },
+        });
+        expect(startDate.format()).toEqual('2019-02-04T00:00:00+11:00');
+        expect(endDate.format()).toEqual('2019-02-04T23:59:59+11:00');
+      });
+    });
+
+    describe('defaultTimePeriod with a non-single period', () => {
+      it('gives nothing by default', () => {
+        const result = getDefaultDates({
+          periodGranularity: 'day',
+          defaultTimePeriod: null,
+        });
+        expect(result).toEqual({});
+      });
+
+      it('basic config works', () => {
+        const { startDate, endDate } = getDefaultDates({
+          periodGranularity: 'day',
+          defaultTimePeriod: {
+            start: { unit: 'day', offset: -1 },
+            end: { unit: 'day', offset: 3 },
+          },
+        });
+        expect(startDate.format()).toEqual('2019-02-04T00:00:00+11:00');
+        expect(endDate.format()).toEqual('2019-02-08T23:59:59+11:00');
+      });
+    });
+  });
+
+  describe('getLimits', () => {
+    it('gives nothing if no config', () => {
+      const { startDate, endDate } = getLimits('day');
+      expect(startDate).toBeNull();
+      expect(endDate).toBeNull();
+    });
+
+    it('throws if either unit does not match period granularity', () => {
+      // The reason for this limitation is to prevent devs from confusing themselves.
+      // E.g. if periodGranularity is one_month_at_a_time, and offset is +5 days, it will not take
+      // effect, the start date will be rounded to the start of the month.
+      const functionCall = () =>
+        getLimits('one_month_at_a_time', { start: { unit: 'day', offset: -3 } });
+      expect(functionCall).toThrow('limit unit must match periodGranularity');
+    });
+
+    it('calculates limits', () => {
+      const { startDate, endDate } = getLimits('day', { start: { unit: 'day', offset: -3 } });
+      expect(startDate.format()).toEqual(moment('2019-02-02T00:00:00').format()); // rounded
+      expect(endDate.format()).toEqual(moment('2019-02-05T23:59:59').format()); // rounded
```

Comment:
Can you do this, like above?
```suggestion
      expect(startDate.format()).toEqual('2019-02-02T00:00:00+11:00'); // rounded
      expect(endDate.format()).toEqual('2019-02-05T23:59:59+11:00'); // rounded
```

Reply:
Updated